src/service/gateway.proto

```diff
@@ -17,6 +20,18 @@ enum close_state {
 
 /* general */
 
+message empty {}
+
+message gateway_success_resp {
+  bytes resp = 1;
+  bytes details = 2;
+}
+
+message gateway_error_resp {
+  bytes error = 1;
```

Comment:
would it be possible to have the error be an actual enum here? it'd document and be more specific about what the possible errors are.

Reply:
yeah i did actually change these but it must be on another branch.  Will fix